quiz.js

```diff
@@ -1,168 +1,137 @@
+    sendApiRequest();
 
     //Api request
     async function sendApiRequest() {
         // set response equal to response from url using fetch method
         let response = await fetch(`https://opentdb.com/api.php?amount=10&type=multiple`);
         // Takes a Response stream and reads it to completion. It returns a promise that resolves with the result of parsing the body text as JSON, which is a JavaScript value of datatype object, string, etc. *** from MDN
         let data = await response.json();
-        //set questions equal to all returned data using the spread operator
         questions = [...data.results];
-        //call generateQuestion function
         generateQuestion();
-        console.log(questions)
     }
-    //this function generates new questions
-    function generateQuestion() {
 
-        //if statement to call endGame function once questions.length is reached
+    //this function generates the next question
+    function generateQuestion() {
+        //If we've gone through all the questions, end the game
         if (questionIndex === questions.length) {
             endGame();
         }
 
-        //emptys answer buttons
-        answerZoneEl.empty();
-        //emptys right/wrong area
-        rightWrongEl.empty();
-
-        cheer.pause();
-        jeer.pause();
-        function fix (errorString){
-            if(errorString.includes("&#039;")){
-                console.log("its in there")
-            }
-            else {(errorString.includes("&#039;"))
-                    console.log("its not there")
-            }
-            while(errorString.includes("&#039;")){
-                errorString = errorString.replace("&#039;", "\'")
-                }
-            while(errorString.includes("&quot;")){
-                errorString = errorString.replace("&quot;","\"")
-                }
-            while(errorString.includes("&amp;")){
-                errorString = errorString.replace("&amp;","&")
-            }
-            while(errorString.includes("&Uuml;")){
-                errorString = errorString.replace("&Uuml;","Ü")
-            }    
-            while(errorString.includes("&eacute;")){
-                errorString = errorString.replace("&eacute;","é")
-            }        
-            return errorString
-        }
-        //sets currentQuestion equal to questions index
+        resetQuestionZone();        
+        
         let currentQuestion = questions[questionIndex];
-        //set correctAnswer equal to the correct answer of the current question
-        correctAnswer = currentQuestion.correct_answer;
-        //tells user what question number they are on
+        correctAnswer       = fix(currentQuestion.correct_answer);
+
         questionNumberEl.text(`Question: ${questionIndex + 1} / 10`)
-        //tells user score
-        scoreEl.text(`Score: ${score}`);
-        //shows user the current question after a fix
-        let fixedCurrentQuestion = fix(currentQuestion.question)
```

Comment:
The approach of `currentQuestion.question = fix(currentQuestion.question);` as done on (the removed) line 93 also works. I prefer that to this line 86 of making a new variable just for the fixed version. Ideally, the api would serve us data we wouldn't have to fix.